### DISCUSSION TITLE
How to define a default expression or qualifiedByName for all field?

### DISCUSSION THREAD

**Discussion by Pumpkinor** [mapstruct/mapstruct#3624] [Q&A] [answered]
Hello，I need some help.
There is one class with many field (like 100 even more) should be convert into another class has same field name, all field need to do some changes from source to target (for example , turn lowcase string into uppercase string)

In the detail, there are class structure:

```java
Obj1.class:
@Data
public class OBJ1 {
    private String property1;
    private String property2;
    ......
    private String property100;
}

Obj2.class:
@Data
public class OBJ2 {
    private String property1;
    private String property2;
    ......
    private String property100;
}
```

```java
@Mapper
public interface OBJTestMapper {

    OBJTestMapper INSTANCE = Mappers.getMapper(OBJTestMapper.class);

    @Mapping(source = "property1", target = "property1", qualifiedByName= "trans")
// or
    @Mapping(source = "property1", target = "property1", expression = "java(StringUtils.upperCase(property1))")
    @Mapping(source = "property2", target = "property2", expression = "java(StringUtils.upperCase(property2))")
    ......
    @Mapping(source = "property100", target = "property100", expression = "java(StringUtils.upperCase(property100))")
   
    OBJ2 merge(OBJ1 obj1);

    @Named("trans")
    static String toId(property) {
        return StringUtils.upperCase(property);
    }
}
```

I konw i can use  @Mapping with expression  or qualifiedByName to achieve my goal.
But  @Mapping should repeat a lot of times, this is somewhat redundant and tiring.
I don't konw does mapstruct support that to to define a default expression or qualifiedByName for all field, but I holp it can do.

**Comment by thunderhook** [comment, marked as answer]
Given your example (and given that all these fields are actually Strings) you can provide a non-qualified mapping method from String to String.
**Everywhere** in your mapper, when you map between a String and another String this method will be used, **unless** you provide a qualifier.

So in your particular case this should do the trick:

```java
@Mapper
public interface OBJTestMapper {

    OBJTestMapper INSTANCE = Mappers.getMapper(OBJTestMapper.class);

    // each property of type String uses "toUpperCase"
    OBJ2 merge(OBJ1 obj1);

    String toUpperCase(String source) {
        return StringUtils.upperCase(source);
    }

}
```

You also don't need to provide the mapping annotations for the fields when they have the same name in source and target.

If you want to exclude certain fields, you can use a named qualifier like this:


```java
@Mapper
public interface OBJTestMapper {

    OBJTestMapper INSTANCE = Mappers.getMapper(OBJTestMapper.class);

    // these 2 properties use "doNothing", everything else uses "toUpperCase"
    @Mapping(target="property50", qualifiedByName="doNothing")
    @Mapping(target="property51", qualifiedByName="doNothing")
    OBJ2 merge(OBJ1 obj1);

    String toUpperCase(String source) {
        return StringUtils.upperCase(source);
    }

    @Named("doNothing")
    String doNothing(String source) {
        return source;
    }
}
```

**Reply by Pumpkinor** [reply]
thanks very much , it helps.
